### DISCUSSION TITLE
Meaning of the code

### DISCUSSION THREAD

**Discussion by jiuker** [minio/minio#15359] [Q&A]
It looks like the logic goes right out inside the loop, but I see that there are two possible validations for the input reference. I checked all the references to this function and there are indeed two places where two validations are used. But the second validation obviously doesn't validate. I don't know if there is a convention for this.
If you pass in two validations and the second one doesn't validate, why do you need to pass it in? If this is a bug, I can fix it. Please let me know what usage is expected.

Definition：`cmd/admin-handler-utils.go:33`
```go
func validateAdminReq(ctx context.Context, w http.ResponseWriter, r *http.Request, actions ...iampolicy.AdminAction) (ObjectLayer, auth.Credentials) {
	// Get current object layer instance.
	objectAPI := newObjectLayerFn()
	if objectAPI == nil || globalNotificationSys == nil {
		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrServerNotInitialized), r.URL)
		return nil, auth.Credentials{}
	}

	for _, action := range actions {
		// Validate request signature.
		cred, adminAPIErr := checkAdminRequestAuth(ctx, r, action, "")
		if adminAPIErr != ErrNone {
			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(adminAPIErr), r.URL)
			return nil, cred
		}
		return objectAPI, cred
	}
	writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrAccessDenied), r.URL)
	return nil, auth.Credentials{}
}
```
Usage：`cmd/admin-handlers-pools.go:131`
```go
func (a adminAPIHandlers) StatusPool(w http.ResponseWriter, r *http.Request) {
	ctx := newContext(r, w, "StatusPool")

	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))

	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.ServerInfoAdminAction, iampolicy.DecommissionAdminAction)
	if objectAPI == nil {
		return
	}
```
Usage：`cmd/admin-handlers-pools.go:169`
```go
func (a adminAPIHandlers) ListPools(w http.ResponseWriter, r *http.Request) {
	ctx := newContext(r, w, "ListPools")

	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))

	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.ServerInfoAdminAction, iampolicy.DecommissionAdminAction)
	if objectAPI == nil {
		return
	}
```

**Comment by jiuker** [comment]
Who can answer my doubts?

**Reply by ravindk89** [reply]
There are no SLAs for discussions or issues. Our engineers have a number of other priorities - they may get to this discussion at any time in the future, or not at all. Bumping the discussion will not increase its priority or position in any queue.

**Comment by klauspost** [comment]
@harshavardhana Hmmm.. It looks like no-one defined what should happen with multiple entries. Is it 'and' or 'or'?

With `iampolicy.ServerInfoAdminAction, iampolicy.DecommissionAdminAction` I assume you mean that you need either, but it seems like only `ServerInfoAdminAction` is checked.

**Reply by harshavardhana** [reply]
Correct @klauspost it needs to skip and move to the next one - it is an 'or' operation.

**Reply by jiuker** [reply]
What are the real expectations? I can contribute. @harshavardhana

**Reply by jiuker** [reply]
The code does not reflect the operation of or @harshavardhana

**Reply by klauspost** [reply]
Sent inhttps://github.com/minio/minio/pull/15372

**Reply by jiuker** [reply]
ok,thx
